sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -396,9 +396,11 @@ case class RangeExec(range: org.apache.spark.sql.catalyst.plans.logical.Range)
     // The default size of a batch, which must be positive integer
     val batchSize = 1000
 
-    val initRangeFuncName = ctx.addNewFunction("initRange",
+    val initRange = ctx.freshName("initRange")
+
+    val initRangeFuncName = ctx.addNewFunction(initRange,
       s"""
-        | private void initRange(int idx) {
+        | private void ${initRange}(int idx) {
```

Comment:
can you add a test? with this bug seems we can't join 2 ranges.

Reply:
@cloud-fan thanks for reviewing. Both `BroadCastExchange` and `ShuffleExchange` don't support `CodegenSupport`, so there should be two `WholeStageCodegen`.

Reply:
Whole-stage codegen is pull-model. I don't think we will have 2 leaf nodes in one stage. Maybe we can just add some comments to explain it and don't change the code.

Reply:
OK, I can just some comments and keep the code unchanged. I changed it here just for better code robustness.

Reply:
Hi @cloud-fan , before adding the comments, I have a question about why we still need `exchange ` if we join two `spark.range(1, 10, 1, 1)`. Because of both of the `range` are only one partition, does the `exchange` really needed?

Reply:
I think you can apply https://github.com/apache/spark/pull/20726 to the `RangeExec` operator and fix this.

Reply:
Thanks for your suggestion, let me take a try.